packages/bot/src/scratch/hooks/flyout_base.js

```diff
@@ -1,78 +1,103 @@
+            Blockly.Events.fire(new Blockly.Events.VarCreate(thisVariable));
+        }
     }
 
-    // Update the scrollbar (if one exists).
-    if (this.scrollbar_) {
-        const newX = x - this.ARROW_SIZE;
+    if (this.autoClose) {
+        this.hide();
+    }
 
-        // Set the scrollbars origin to be the top left of the flyout.
-        this.scrollbar_.setOrigin(newX, y);
-        this.scrollbar_.resize();
+    Blockly.derivWorkspace.toolbox_.clearSelection();
+    newBlock.isInFlyout = false;
 
-        // Set the position again so that if the metrics were the same (and the
-        // resize failed) our position is still updated.
-        this.scrollbar_.setPosition_(this.scrollbar_.position_.x, this.scrollbar_.position_.y);
-    }
+    return newBlock;
 };
 
 /**
- * Get the width of the flyout.
- * @return {number} The width of the flyout.
- * deriv-bot: Return actual width rather than this.DEFAULT_WIDTH.
+ * Copy a block from the flyout to the workspace and position it correctly.
+ * @param {!Blockly.Block} oldBlock The flyout block to copy.
+ * @return {!Blockly.Block} The new block in the main workspace.
+ * @private
  */
-Blockly.Flyout.prototype.getWidth = function() {
-    return this.width_;
+Blockly.Flyout.prototype.placeNewBlock_ = function(event, oldBlock) {
+    const targetWorkspace = this.targetWorkspace_;
+    const svgRootOld = oldBlock.getSvgRoot();
+
+    if (!svgRootOld) {
+        throw new Error('oldBlock is not rendered.');
+    }
+  
+    // Create the new block by cloning the block in the flyout (via XML).
+    const xml = Blockly.Xml.blockToDom(oldBlock);
+    // The target workspace would normally resize during domToBlock, which will
+    // lead to weird jumps.  Save it for terminateDrag.
+    targetWorkspace.setResizesEnabled(false);
+  
+    // Using domToBlock instead of domToWorkspace means that the new block will be
+    // placed at position (0, 0) in main workspace units.
+    const block = Blockly.Xml.domToBlock(xml, targetWorkspace);
+    const svgRootNew = block.getSvgRoot();
+
+    if (!svgRootNew) {
+        throw new Error('block is not rendered.');
+    }
+  
+    // The offset in pixels between the main workspace's origin and the upper left
+    // corner of the injection div.
+    const mainOffsetPixels = targetWorkspace.getOriginOffsetInPixels();
+  
+    // The offset in pixels between the flyout workspace's origin and the upper
+    // left corner of the injection div.
+    const flyoutOffsetPixels = this.workspace_.getOriginOffsetInPixels();
+  
+    // The position of the old block in pixels relative to the flyout
+    // workspace's origin.
+    const oldBlockPosPixels = new goog.math.Coordinate(event.clientX - 50 , event.clientY - 20);
+  
+    // The position of the old block in pixels relative to the upper left corner
+    // of the injection div.
+    const oldBlockOffsetPixels = goog.math.Coordinate.sum(flyoutOffsetPixels, oldBlockPosPixels);
+  
```

Comment:
@aaron-binary nitpick: I don't think you need a new line after every definition not sure though.

Reply:
Let's keep it this way as it's more readable.